### PR TITLE
#28846 implementation

### PR DESCRIPTION
**Pull request by ahf**
These patches does the following:

1. Make the `LOG` message use the new K/V parser and the `config_line_t` encoding format.
2. Add the `STATUS` message.

### CODE REVIEW

File: src/feature/client/transports.c [hunk 1]
```diff
@@ -1144,22 +1146,63 @@ parse_log_line(const char *line, managed_proxy_t *mp)
...
     goto done;
   }
 
-  const char *message = line + strlen(PROTO_LOG) + 1;
+  const char *data = line + strlen(PROTO_LOG) + 1;
+  values = kvline_parse(data, KV_QUOTED);
+
+  if (! values) {
+    log_warn(LD_PT, "Managed proxy \"%s\" wrote an invalid LOG message: %s",
+             mp->argv[0], data);
```
Comment (nmathewson) [inline src/feature/client/transports.c]: We should probably use "escaped(data)" herfe.

File: src/feature/client/transports.c [hunk 2]
```diff
@@ -1205,6 +1210,55 @@ parse_log_line(const char *line, managed_proxy_t *mp)
...
+    goto done;
+  }
+
+  /* We check if we received the TYPE parameter, which is the only *required*
+   * value. */
+  const config_line_t *type = config_line_find(values, "TYPE");
+
+  if (! type) {
+    log_warn(LD_PT, "Managed proxy \"%s\" wrote a STATUS line without "
+                    "TYPE: %s", mp->argv[0], data);
```
Comment (nmathewson) [inline src/feature/client/transports.c]: here too.

Comment (ahf) [reply src/feature/client/transports.c]: Fixed in ab0d7d2dd470a447535eaf7c35bd4877c43bce26

File: src/feature/client/transports.c (hunk 1 above)
Comment (ahf) [reply src/feature/client/transports.c]: Fixed in ab0d7d2dd470a447535eaf7c35bd4877c43bce26

File: src/feature/client/transports.c [hunk 3]
```diff
@@ -1205,6 +1210,55 @@ parse_log_line(const char *line, managed_proxy_t *mp)
...
+    goto done;
+  }
+
+  const char *data = line + strlen(PROTO_STATUS) + 1;
+
+  values = kvline_parse(data, KV_QUOTED);
+
+  if (! values) {
+    log_warn(LD_PT, "Managed proxy \"%s\" wrote an invalid "
+             "STATUS message: %s", mp->argv[0], data);
```
[the remark this answers is not in the document]
Comment (ahf) [reply src/feature/client/transports.c]: Fixed in ab0d7d2dd470a447535eaf7c35bd4877c43bce26

**Pull request merged**